### ISSUE TITLE
General distributed training query

### ISSUE THREAD

**Issue by aakash-saboo** [spellrun/spell-deeplab-voc-2012#3]
Hi
I see that the learning rate is multiplied by world size. Is it because in distributed training, the gradients as SUMMED up rather than averaged as done in DataParallel mode?

**Comment by ResidentMario** [comment]
No, it's just a rule-of-thumb.

If your batch size is 8, a learning rate of 0.001 might be appropriate.

If your batch size is 32, you might be able to choose a larger initial learning rate, because your gradient steps will be less random.

The world size determines the effective batch size (e.g. 4 GPUs and a batch size of 8 per GPU is an effective batch size of 32), so multiplying `lr` by world size gets us a better (larger) initial learning rate (in this example, one of `0.001 * 4 = 0.004`).

**Comment by aakash-saboo** [comment]
Okay.
But the fact that the gradients are summed up instead of being averaged (as
done in data parallel) is indeed correct right?

**Comment by ResidentMario** [comment]
I don't think so. I think the both `DistributedDataParallel` (PyTorch) and `DistributedOptimizer` (Horovod) perform an average by default:

```
    optimizer = hvd.DistributedOptimizer(optimizer,
                                         named_parameters=model.named_parameters(),
                                         compression=compression,
                                         op=hvd.Average)
```

I'm pretty sure they can both be configured to do a sum instead, if you'd like. But that shouldn't be the current behavior in these scripts as written.
